dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -739,148 +768,213 @@ public String getWorkflowDag(@ApiParam(value = "workflowId", required = true) @P
+                                        if(type.equals("tools")){
+                                            //get the docker file and link
+                                            dockerPullURL = getURLFromEntry((String)requirement.get("dockerPull"));
+
+                                            //put the tool ID and docker information into two different maps
+                                            toolID.put(index.toString(), new MutablePair<>(step.get("id").toString(), fileName));
+                                            toolDocker.put(index.toString(),new MutablePair<>(defaultDockerEnv, dockerPullURL));
+                                            index++;
+                                        }else{
+                                            nodePairs.add(new MutablePair<>(step.get("id").toString().replaceFirst("#", ""), getURLFromEntry(requirement.get("dockerPull").toString())));
+                                        }
                                         defaultDocker = false;
                                         break;
                                     }
                                 }
                             }
                         }
+
                         if (defaultDocker) {
-                            nodePairs.add(new MutablePair<>(step.get("id").toString().replaceFirst("#", ""), getURLFromEntry(defaultDockerEnv)));
+                            if(type.equals("tools")) {
+                                // no docker requirement, put the content as blank values
+                                Map<String, Pair<String, String>> dockerPull = new HashMap<>(); // this will be inserted into toolDocker
+
+                                if(defaultDockerEnv.equals("")){
+                                    defaultDockerEnv = "Not Specified";
+                                    dockerPullURL = "Not Specified"; // the workflow does not specify any docker requirement too
+                                }else{
+                                    dockerPullURL = getURLFromEntry(defaultDockerEnv); //get default from workflow docker requirement
+                                }
+
+                                toolID.put(index.toString(), new MutablePair<>(step.get("id").toString(), fileName));
+                                toolDocker.put(index.toString(), new MutablePair<>(defaultDockerEnv, dockerPullURL));
+                                index++;
+                            }else{
+                                nodePairs.add(new MutablePair<>(step.get("id").toString().replaceFirst("#", ""), getURLFromEntry(defaultDockerEnv)));
+                            }
                         }
                     }
                 }
             }
         } catch (FileNotFoundException e) {
             e.printStackTrace();
```

Comment:
Again, don't just throw exceptions away. 
http://digitaldrummerj.me/dont-swallow-the-exceptions/

Instead, log the exception and then in this case, you can throw an exception to inform the UI that something went wrong. 

ex:
```
LOG.error("Could not read WDL file");
throw new CustomWebApplicationException("Could not read WDL file", HttpStatus.SC_INTERNAL_ERROR);
``